src/test/java/uk/gov/hmcts/juror/api/moj/report/grouped/UnpaidAttendanceSummaryReportTest.java

```diff
@@ -0,0 +1,170 @@
+    }
+
+    @BeforeEach
+    @Override
+    public void beforeEach() {
+        this.securityUtilMockedStatic = mockStatic(SecurityUtil.class);
+        this.courtLocationService = mock(CourtLocationService.class);
+        super.beforeEach();
+    }
+
+    @AfterEach
+    void afterEach() {
+        securityUtilMockedStatic.close();
+    }
+
+    @Override
+    public UnpaidAttendanceSummaryReport createReport(PoolRequestRepository poolRequestRepository) {
+        return new UnpaidAttendanceSummaryReport(poolRequestRepository, this.courtLocationService);
+    }
+
+    @Override
+    protected StandardReportRequest getValidRequest() {
+        return StandardReportRequest.builder()
+            .reportType(report.getName())
+            .fromDate(LocalDate.of(2020, 1, 1))
+            .toDate(LocalDate.of(2020, 1, 2))
+            .build();
+    }
+
+    @Override
+    @DisplayName("positivePreProcessQueryTypicalCourt")
+    public void positivePreProcessQueryTypical(JPAQuery<Tuple> query, StandardReportRequest request) {
+        request.setToDate(LocalDate.of(2023, 1, 1));
+        request.setToDate(LocalDate.of(2023, 1, 2));
+        securityUtilMockedStatic.when(SecurityUtil::isCourt).thenReturn(true);
+        securityUtilMockedStatic.when(SecurityUtil::getActiveOwner).thenReturn(TestConstants.VALID_COURT_LOCATION);
+        report.preProcessQuery(query, request);
+
+        verify(query).where(
+            QAppearance.appearance.attendanceDate.between(request.getFromDate(), request.getToDate())
+                .and(QAppearance.appearance.appearanceStage.eq(AppearanceStage.EXPENSE_ENTERED))
+                .or(QAppearance.appearance.appearanceStage.eq(AppearanceStage.EXPENSE_EDITED))
+                .and(QAppearance.appearance.isDraftExpense.isFalse()));
+
+        verify(query)
+            .orderBy(QPool.pool.poolNumber.asc(), QJuror.juror.jurorNumber.asc());
+    }
+
+    @Override
+    @DisplayName("positiveGetHeadingsTypicalCourt")
+    @SuppressWarnings("unchecked")
+    public Map<String, AbstractReportResponse.DataTypeValue> positiveGetHeadingsTypical(
+        StandardReportRequest request,
+        AbstractReportResponse.TableData<GroupedTableData> tableData,
+        GroupedTableData data) {
+
+        when(request.getFromDate()).thenReturn(LocalDate.of(2023, 3, 1));
+        when(request.getToDate()).thenReturn(LocalDate.of(2023, 3, 2));
+        when(request.getLocCode()).thenReturn("415");
+        when(data.getSize()).thenReturn(2);
+        securityUtilMockedStatic.when(SecurityUtil::isCourt).thenReturn(true);
+        securityUtilMockedStatic.when(SecurityUtil::getActiveOwner).thenReturn(TestConstants.VALID_COURT_LOCATION);
+
+        CourtLocation courtLocation = mock(CourtLocation.class);
+        when(courtLocationService.getCourtLocation(TestConstants.VALID_COURT_LOCATION)).thenReturn(courtLocation);
+        doReturn(getCourtNameEntry()).when(report).getCourtNameHeader(courtLocation);
```

Comment:
I don't think this is doing anything as the root object is not mocked (I could be wrong)?

Reply:
Removed

Reply:
test returns null when removed so is required